Tests/LotsawaTests/AdjacencyMatrixTests.swift

```diff
@@ -2,66 +2,75 @@
 import XCTest
 
 class AdjacencyMatrixTests: XCTestCase {
-  func testEmpty() {
-    let x = AdjacencyMatrix()
-    for i in 0..<64 {
-      for j in 0..<64 {
-        XCTAssert(!x.hasEdge(from: i, to: j))
+
+  /// The range of values used to construct matrices in these tests.
+  let universe = 1..<(Int.bitWidth * 2)
+  
+  /// Constructs an AdjacencyMatrix with an edge from `i` to `j` iff `fromPredicate(i, j)`.
+  func makeAdjacencyMatrix(fromPredicate predicate: (Int, Int) -> Bool) -> AdjacencyMatrix {
+    var result = AdjacencyMatrix();
+    for i in universe {
+      for j in universe where predicate(i, j) {
+          result.addEdge(from: i, to: j);
       }
     }
+    return result;
   }
-
-  let universe = 1..<(Int.bitWidth * 2)
-  func edgeShouldExist(from x: Int, to y: Int) -> Bool { x != y && x % y == 0 }
-
-  func divisibility() -> AdjacencyMatrix {
-    var a = AdjacencyMatrix()
-
-    for x in universe {
-      for y in universe {
-        if edgeShouldExist(from: x, to: y) {
-          a.addEdge(from: x, to: y)
-        }
+  
+  /// Tests that an empty AdjacencyMatrix has no edges.
+  func testEmpty() {
+    let x = AdjacencyMatrix()
+    for i in universe {
+      for j in universe {
+        XCTAssert(!x.hasEdge(from: i, to: j))
       }
     }
-    return a
   }
-
-  func testAddEdge() {
-    let a = divisibility()
-
-    for x in universe {
-      for y in universe {
-        XCTAssertEqual(a.hasEdge(from: x, to: y), edgeShouldExist(from: x, to: y), "\(x) -> \(y)")
+  
+  /// Returns true iff `y` divides `x` and `x` and `y` are inequal.
+  let nontriviallyDivides = { (_ x: Int, _ y: Int) in x != y && x % y == 0 };
+  
+  /// Ensures an `AdjacencyMatrix` modeling integer divisibility has the expected edges.
+  func testDivisibility() {
+    let divisibilityMatrix = makeAdjacencyMatrix(fromPredicate: nontriviallyDivides);
+    for i in universe {
+      for j in universe {
+        XCTAssertEqual(divisibilityMatrix.hasEdge(from: i, to: j), nontriviallyDivides(i, j), "\(i) -> \(j)")
       }
     }
   }
 
+  /// Ensures an `AdjacencyMatrix` modeling integer divisibility
   func testTransitiveClosure() {
-    var a = divisibility()
-    a.formTransitiveClosure()
+    var divisibilityMatrix = makeAdjacencyMatrix(fromPredicate: nontriviallyDivides);
```

Comment:
Why is "Matrix" in this name?  Logically, it's a graph in adjacency matrix form.  Or it's a relation, or something.

Reply:
Good point. Perhaps it should return to just `divisibility`.